src/Common/src/Interop/Unix/System.Native/Interop.MessageHeader.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal unsafe struct IOVector
+        {
+            public byte* Base;
+            public UIntPtr Count;
```

Comment:
While that's the correct tradeoff for many/most situations, it is not so here: in this case, I have specifically chosen `uintptr_t`/`UIntPtr` to avoid the aforementioned allocation + copying. Throughout the rest of the shim, I've used `uint64_t`/`ulong` in place of `size_t` (or `int64_t`/`long` in place of `ssize_t`).